store/searchlayer/layer.go

```diff
@@ -18,6 +20,7 @@ type SearchStore struct {
 	channel      *SearchChannelStore
 	post         *SearchPostStore
 	config       *model.Config
+	configLock   sync.RWMutex
```

Comment:
Also a `sync.Mutex` here will be just fine.

Reply:
I am wondering if using `atomic.Value` would be more appropriate in this case. We use that consistently in the `app.Server` struct.